File: 2508/2508.c
int addEdgesToMakeDegreesOfAllNode(struct TreeNode* root) {
    if (!root) return 0;
    int l = addEdgesToMakeDegreesOfAllNode(root->left);
    int r = addEdgesToMakeDegreesOfAllNode(root->right);
    return 1 + (l > r ? l : r);
}

File: 2508/2508.cs
public class Solution {
    public int AddEdgesToMakeDegreesOfAllNode(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(AddEdgesToMakeDegreesOfAllNode(root.left), AddEdgesToMakeDegreesOfAllNode(root.right));
    }
}